backend/dataall/modules/metadata_forms/services/metadata_form_service.py

```diff
@@ -35,8 +36,46 @@ def validate_create_field_params(data):
         if 'displayNumber' not in data:
             raise exceptions.RequiredParameter('displayNumber')
 
-        if data.get('type') == MetadataFormFieldType.GlossaryTerm.value and 'glossaryNodeUri' not in data:
-            raise exceptions.RequiredParameter('glossaryNodeUri')
+        if data.get('type') == MetadataFormFieldType.GlossaryTerm.value:
+            if 'glossaryNodeUri' not in data:
+                raise exceptions.RequiredParameter('glossaryNodeUri')
+            MetadataFormParamValidationService.validate_glossary_node_uri(data.get('glossaryNodeUri'))
+        else:
+            MetadataFormParamValidationService.validate_field_possible_values_params(data)
+
+    @staticmethod
+    def validate_update_field_params(form_uri, data):
+        if data.get('metadataFormUri') != form_uri:
+            raise Exception('property metadataFormUri does not match form uri')
+
+        if 'displayNumber' not in data:
+            raise exceptions.RequiredParameter('displayNumber')
+
+        if data.get('type') == MetadataFormFieldType.GlossaryTerm.value:
+            if 'glossaryNodeUri' not in data:
+                raise exceptions.RequiredParameter('glossaryNodeUri')
+            MetadataFormParamValidationService.validate_glossary_node_uri(data.get('glossaryNodeUri'))
+        else:
+            MetadataFormParamValidationService.validate_field_possible_values_params(data)
+
+    @staticmethod
+    def validate_glossary_node_uri(uri):
+        with get_context().db_engine.scoped_session() as session:
+            try:
+                GlossaryRepository.get_node(session, uri)
+                return True
+            except exceptions.ObjectNotFound:
+                raise exceptions.InvalidInput('glossaryNodeUri', uri, 'from glossary list')
+
+    @staticmethod
+    def validate_field_possible_values_params(data):
```

Comment:
general q: should we not allow possible values for Boolean types since it can only be T/F

Reply:
I see this is implemented on FE - I guess minor nit if we should even save possibleValues to DB on backend for Boolean types? (thinking of case with programmatic access and specifying possible Values still for boolean types)

Reply:
[8bbefd2](https://github.com/data-dot-all/dataall/pull/1480/commits/8bbefd22389d62a137fe26de245f7f2bbea15a7a)

Reply:
neat!